### PR TITLE
continued 3.3.4 development

### PR DESCRIPTION
**Pull request by drenalin23**
## What
Add additional attributes for filter-api-validator recipe.  The API Validation filter has several variables we don't include - this finished optional attributes for the rate-limiting element.
Set the multi-role-match attribute of the validators element to be optional, it will be deprecated in the upcoming Repose v8.

## How
Updated templates, recipes and default attributes.

### CODE REVIEW

File: templates/default/rate-limiting.cfg.xml.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <% if !@version.nil? && @version.split('.')[0].to_i < 7 %>
 <rate-limiting xmlns="http://docs.rackspacecloud.com/repose/rate-limiting/v1.0">
 <% else %>
-<rate-limiting xmlns="http://docs.openrepose.org/repose/rate-limiting/v1.0">
+<rate-limiting <% if @datastore %><datastore="<% @datastore %>"> <% end ><% if @datastore_warn_limit %><datastore-warn-limit="<% @datastore_warn_limit %>"> <% end %><% if @overlimit_429_responsecode %><overlimit-429-responseCode="<% @overlimit_429_responsecode %>"> <% end %><% if @use_capture_groups %><use-capture-groups="<% @use_capture_groups %>"> <% end %><xmlns="http://docs.openrepose.org/repose/rate-limiting/v1.0">
```
Comment (usnavi) [inline templates/default/rate-limiting.cfg.xml.erb]: Is this right, is datastore an attribute?  E.g., https://github.com/rackerlabs/repose/blob/master/repose-aggregator/components/filters/rate-limiting/src/main/resources/META-INF/schema/examples/rate-limiting.cfg.xml

Comment (usnavi) [inline templates/default/rate-limiting.cfg.xml.erb]: I think all those attributes just need a starting quote characters

Comment (drenalin23) [inline templates/default/rate-limiting.cfg.xml.erb]: Yes!  Good catch.  I removed all of the unnecessary carrots on that line and looked ahead and noticed missing quotes in validator template for the attribute I made optional in that template.

**Comment by shintasmith** [comment]
lgtm 👍

**Pull request merged**